edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/authenticators/DeviceScopeCertificateAuthenticator.cs

```diff
@@ -56,48 +47,11 @@ protected override bool ValidateWithServiceIdentity(ServiceIdentity serviceIdent
             }
             else if (serviceIdentity.Authentication.Type == ServiceAuthenticationType.CertificateThumbprint)
             {
-                result = CertificateHelper.ValidateClientCert(certificateCredentials.ClientCertificate,
-                                                              certificateCredentials.ClientCertificateChain,
-                                                              Option.None<IList<X509Certificate2>>(),
-                                                              Events.Log);
-                if (!result)
-                {
-                    Events.InvalidCertificate(serviceIdentity.Id, certificateCredentials);
-                }
-                else
-                {
-                    result = serviceIdentity.Authentication.X509Thumbprint.Map(
-                        t =>
-                        {
-                            List<string> thumbprints = new List<string>() { t.PrimaryThumbprint, t.SecondaryThumbprint };
-                            return CertificateHelper.ValidateCertificateThumbprint(certificateCredentials.ClientCertificate, thumbprints);
-                        })
-                        .GetOrElse(() => throw new InvalidOperationException($"Unable to validate certificate because the service identity has empty thumbprints"));
-                    if (!result)
-                    {
-                        Events.ThumbprintMismatch(serviceIdentity.Id);
-                    }
-                }
+                result = this.ValidateThumbprintAuth(serviceIdentity, certificateCredentials);
```

Comment:
I see there is difference then in how DPS and IoT Hub authenticate certs. In that case what you have done here seems reasonable.

Reply:
It is possible, I didn't check DPS, but Rajeev said even DPS does the same as IoTHub.

Reply:
I reviewed the code in both services and they are different but for the DPS it is okay to enforce this and for IoT Hub to be a bit more permissive. So we can call this resolved.